archive-tar.c

```diff
@@ -38,11 +37,18 @@ static int write_tar_filter_archive(const struct archiver *ar,
 #define USTAR_MAX_MTIME 077777777777ULL
 #endif
 
+/* writes out the whole block */
+static void write_block(const char * block) {
```

Comment:
It might make sense to call it `write_block_or_die()` and use `die()` instead of `error()` in case `gzwrite()` fails. That would make the `gzip` code more consistent with the `!gzip` one.